readthedocs/rtd_tests/tests/test_footer.py

```diff
@@ -11,31 +13,38 @@
 from readthedocs.builds.constants import BRANCH, LATEST, TAG
 from readthedocs.builds.models import Version
 from readthedocs.core.middleware import FooterNoSessionMiddleware
+from readthedocs.projects.constants import PUBLIC
 from readthedocs.projects.models import Project
 
 
-class Testmaker(APITestCase):
-    fixtures = ['test_data']
-    url = '/api/v2/footer_html/?project=pip&version=latest&page=index&docroot=/'
-    factory = APIRequestFactory()
+class TestFooterHTML(TestCase):
+
+    def setUp(self):
+        self.pip = get(
+            Project,
+            slug='pip',
+            repo='https://github.com/rtfd/readthedocs.org',
+            privacy_level=PUBLIC,
+            main_language_project=None,
+        )
+        self.latest = self.pip.versions.get(slug=LATEST)
+        self.url = (
+            reverse('footer_html') +
+            f'?project={self.pip.slug}&version={self.latest.slug}&page=index&docroot=/'
+        )
 
-    @classmethod
-    def setUpTestData(cls):
-        cls.pip = Project.objects.get(slug='pip')
-        cls.latest = cls.pip.versions.create_latest()
+        self.factory = APIRequestFactory()
 
     def render(self):
-        request = self.factory.get(self.url)
-        response = FooterHTML.as_view()(request)
-        response.render()
-        return response
+        r = self.client.get(self.url)
+        return r
 
     def test_footer(self):
         pip = Project.objects.get(slug='pip')
         pip.show_version_warning = True
         pip.save()
 
-        r = self.client.get(self.url)
+        r = self.render()
```

Comment:
I would avoid wrapping a well known function call (`self.client.get(url)`) into something non-standard (`self.render`) just to save some chars. This just makes the code harder to read only.

Reply:
Also, `render` in a Django context, means rendering a template and returning a HttpResponse.